py-polars/polars/datatypes/convert.py

```diff
@@ -150,7 +151,8 @@ def _map_py_type_to_dtype(
     raise TypeError("invalid type")
 
 
-def is_polars_dtype(dtype: Any, *, include_unknown: bool = False) -> bool:
+@functools.lru_cache
```

Comment:
Can you set a reasonable upper bound.

Reply:
`functools.lru_cache` uses a default `maxsize` of 128.  That should be fine, especially since the local `cache()` decorator function also sticks with that default. Let me know, if you think otherwise.

Reply:
> `functools.lru_cache` uses a default `maxsize` of 128. That should be fine, especially since the local `cache()` decorator function also sticks with that default. Let me know, if you think otherwise.

I'd still make it explicit, just so it's clear in the code.